### DISCUSSION TITLE
Codespaces unavailable for authorized GitHub account

### DISCUSSION THREAD

**Discussion by valhuber** [community/community#32791] [Codespaces]
In most cases, the Code -> Codespaces tab is visible and enabled.   But for [this account](https://github.com/KatrinaHuberJuma) trying to load [this project](https://github.com/ApiLogicServer/ApiLogicProject) into Codespaces,, it's not visible:

![no codespaces](https://user-images.githubusercontent.com/5579631/189694112-eb3ef152-7f72-42ff-b6eb-021ae7d84f49.jpeg)

**Comment by dmilburn** [comment]
Hi @valhuber, thanks for reaching out! I looked into it and `KatrinaHuberJuma` is not part of the codespaces private beta. It looks like the organization hasn't enabled codespaces. Anyone who is in the codespaces individual beta can still create codespaces on this repo since it's public but since they're not, they are unable to see it.

This should be fixable just by enabling codespaces for the organization: https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/enabling-github-codespaces-for-your-organization

Let me know if you need more help!

**Reply by valhuber** [reply]
I have spoken to the Product Manager (Craig Peters), who has recently altered it so the Beta is not required... if you have a GitHub account, you should be in. This has in fact been true for several colleagues.

Katrina (my daughter!) is probably in an org that has not gone to CS, but, I think it's quite important that even org folk be able to open CS.  As I explained to Craig, this is a FANTASTIC way to **_evaluate_** software (eg, open source [API Logic Server](https://github.com/ApiLogicServer/ApiLogicProject)).  I would think Org folks would be a _high_ priority for such support, since trying things might lead to adoption of CS.

By contrast, asking eval folks to get their company to sign up for CS is a big ask, in fact I'd think highly unlikely.

BTW, her org is a boot camp, another good target for you product (I would think).

**Reply by dmilburn** [reply]
@valhuber, thanks for the feedback! I'm so glad codespaces has been helpful for you in evaluating software!

I heard you were previously in contact with Craig and I see that API Logic Server is on a free codespaces plan. I know this isn't your larger point, but if you were to enable codespace creation at the org level for API Logic Server, Katrina would be able to make codespaces on that repo. But to avoid future confusion, I've added Katrina to the Codespaces individual beta so she should have access regardless now.

I totally hear you on your larger point of the challenge of evaluating a new tool when organization bureaucracy and risk is involved. Codespaces access overall is something we're re-evaluating and will hopefully have a good update on that soon.

> BTW, her org is a boot camp, another good target for you product (I would think).

If her boot camp would be interested in some kind of larger Codespaces use/program, I would get in contact with GitHub support directly about that or have her open a discussion here!

**Reply by valhuber** [reply]
@dmilburn - thanks for getting back.

It is super-urgent for me to ensure that anyone, *especially org folks,* can access API Logic Server for eval.  Otherwise, it's not an oval path...

I was not aware of "enable codespace creation at the org level for API Logic Server" per your note above... Is the screen shot below what you mean?  I am unclear on the implications of "access data" - I just want them to be able to "load" a project in Codespaces...

<img width="1104" alt="3rd party access" src="https://user-images.githubusercontent.com/5579631/189937188-7b6a31a5-6fe6-4d6b-8cdf-00a350c2ef32.png">

**Reply by dmilburn** [reply]
Hi @valhuber, did you try following the docs link here? https://github.com/orgs/community/discussions/32791#discussioncomment-3629258

I believe choosing the "All members" option [here](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/enabling-github-codespaces-for-your-organization#choose-who-can-create-codespaces-that-are-billed-to-your-organization) will fix the problem for all members of your org

**Reply by valhuber** [reply]
@dmilburn 

Odd... I activated that as you suggest, and it worked for Katrina.

But [Niteish in India](https://github.com/nitheish-ss) tried it and it failed:

![no-codespaces](https://user-images.githubusercontent.com/5579631/190195026-65dd59ae-93f5-456e-98af-0ee26702af6e.png)

I am nearly ready to make this the default eval, but I cannot do that unless these authorization issues don't occur...

**Reply by dmilburn** [reply]
@valhuber in order for someone to have access to codespaces on your repo, with the current setting, they need to be added as a member to your organization. I just checked and Niteish is not an organization member nor a collaborator which is why they are unable to create codespaces on your repo. 

The current setting restricts codespaces to organization members. If you don't want to make them a full member, you can make them an outside collaborator instead and change the setting [here](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/enabling-github-codespaces-for-your-organization#choose-who-can-create-codespaces-that-are-billed-to-your-organization) to "All members and outside collaborators"

Here's some more documentation on the distinction between member and collaborator:
- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-teams-and-people-with-access-to-your-repository#about-access-management-for-repositories
- https://docs.github.com/en/organizations/managing-access-to-your-organizations-repositories/converting-an-organization-member-to-an-outside-collaborator

**Reply by valhuber** [reply]
@dmilburn 

Yes, you told me that a while ago (thanks), and I set it.  

It worked for Katrina, but not Niteish.  Are there perhaps "eventual consistency" issues at play here, and things have not migrated to India?

<img width="1079" alt="all-member-already-set" src="https://user-images.githubusercontent.com/5579631/190670775-1581d758-bc4f-40f4-8905-d850866b3f4a.png">

**Reply by dmilburn** [reply]
@valhuber Sorry for being unclear, the issue here is that *Niteish* is not an organization member

**Comment by valhuber** [comment]
@dmilburn 

I have tried it with 2 other non-org users, and it has worked.  For example, here it is running with my wife's account - gloria.gordon.huber@gmail.com - not an org member:

![codespaces-enabled](https://user-images.githubusercontent.com/5579631/190685114-affbcd39-24ca-45f5-bf7a-a052a4e03ec5.png)

The key goal here is that _any_ GitHub user - _org or no org_ - be able to run the eval.  I
cannot pre-authorize, or guess who is / is not in some org, no?

On Fri, Sep 16, 2022 at 8:31 AM Debbie Milburn ***@***.***>
wrote:

> @valhuber <https://github.com/valhuber> Sorry for being unclear, the
> issue here is that *Nitesh* is not an organization member
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/community/discussions/32791#discussioncomment-3663816>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABKSG3Z53W2TC7ACHQHQYNDV6SHDNANCNFSM6AAAAAAQKSXKNQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 
Thanks,
Val

510-703-5909  valhuber on skype

**Reply by craiglpeters** [reply]
This will no longer be an issue later in the year when we make Codespaces available to all GitHub users. In the mean time, as a participant in our limited preview for open source, only others who have access to Codespaces independent or whom you give access as members or collaborators in your organization will be able to create Codespaces

**Reply by valhuber** [reply]
@craiglpeters @dmilburn 

Oh... I interpret _"only beta users + authorized members/collaborations"_ to mean it is not suitable for evaluation at this point?  I mistook the successful tests to indicate general access was now enabled, so I guess you must have authorized some specific folks for Beta?

Please confirm, since I started down the path of making this the [recommended eval](https://valhuber.github.io/ApiLogicServer/Install-Eval/) - it now seems I should roll that back, right?

**Reply by dmilburn** [reply]
@valhuber yes, apologies for the miscommunication. Later in the year Codespaces will be available for all but for now you must be in an organization with codespaces enabled or be a part of the beta.
